flang/lib/Optimizer/Transforms/AddAliasTags.cpp

```diff
@@ -57,12 +62,131 @@ static llvm::cl::opt<unsigned> localAllocsThreshold(
 
 namespace {
 
+// Return the size and alignment (in bytes) for the given type.
+// TODO: this must be combined with DebugTypeGenerator::getFieldSizeAndAlign().
+// We'd better move fir::LLVMTypeConverter out of the FIRCodeGen component.
+static std::pair<std::uint64_t, unsigned short>
+getTypeSizeAndAlignment(mlir::Type type,
+                        fir::LLVMTypeConverter &llvmTypeConverter) {
+  mlir::Type llvmTy;
+  if (auto boxTy = mlir::dyn_cast_if_present<fir::BaseBoxType>(type))
+    llvmTy = llvmTypeConverter.convertBoxTypeAsStruct(boxTy, getBoxRank(boxTy));
+  else
+    llvmTy = llvmTypeConverter.convertType(type);
```

Comment:
Type conversion fails for `!fir.boxproc` members of derived types, e.g. for:
```
  Type t
    Sequence
    Procedure(),Pointer,NoPass :: sub_ptr
    Integer i
  End Type
```

I wonder if I should return a pointer type in `LLVMTypeConverter::LLVMTypeConverter`: 
```
  addConversion([&](BoxProcType boxproc) {
    // TODO: Support for this type will be added later when the Fortran 2003
    // procedure pointer feature is implemented.
    return std::nullopt;
  });
```

Or, is there a better solution?

This issue appeared in just one test in my local testing, but I cannot merge this without resolving the problem.

Reply:
It looks like other compilers (NVHPC/gfortran/ifx) do support sequence derived types with procedure pointer components inside COMMON blocks, so it looks like it might be some useful extension of the standard.

I think I am going to try to schedule `AddAliasTags` pass after the `BoxedProcedurePass` inside `createDefaultFIRCodeGenPassPipeline`. I hope this okay with you (given that this is one of the alternatives you listed :) )